VMEncryption/main/handle.py

```diff
@@ -61,16 +61,23 @@
 
 def install():
     hutil.do_parse_context('Install')
+    # The extension update handshake is [old:disable][new:update][old:uninstall][new:install]
+    # Prior extensions archived their configs in [old:uninstall], not in [old:disable]
+    # As such, the first opportunity to restore old configs is in [new:install]
+    # In the future, moving the config archive/restore step to disable/update is preferred 
     hutil.restore_old_configs()
     hutil.do_exit(0, 'Install', CommonVariables.extension_success_status, str(CommonVariables.success), 'Install Succeeded')
 
 def disable():
     hutil.do_parse_context('Disable')
+    # archiving old configs during disable rather than uninstall will allow subsequent versions 
+    # to restore these configs in their update step rather than their install step once all 
+    # released versions of the extension are at this version or above 
+    hutil.archive_old_configs()
```

Comment:
Did we actually figure out why we even archive old configs?

Reply:
If the most recent configuration settings file does not include encryption operation and volume type information, the extension will walks backwards based on file timestamp through the archived settings file until it finds the setting file that contains this information.